runtime/src/tasks/qthreads/tasks-qthreads.c

```diff
@@ -913,36 +913,32 @@ chpl_taskID_t chpl_task_getId(void)
 
 void chpl_task_sleep(double secs)
 {
-    if (qthread_shep() == NO_SHEPHERD) {
-        struct timeval deadline;
-        struct timeval now;
-
-        //
-        // Figure out when this task can proceed again, and until then, keep
-        // yielding.
-        //
-        gettimeofday(&deadline, NULL);
-        deadline.tv_usec += (suseconds_t) ((secs - trunc(secs)) * 1.0e6);
-        if (deadline.tv_usec > 1000000) {
-            deadline.tv_sec++;
-            deadline.tv_usec -= 1000000;
-        }
-        deadline.tv_sec += (time_t) trunc(secs);
+    qthread_shepherd_id_t shep = qthread_shep();
+    struct timeval deadline;
+    struct timeval now;
+
+    //
+    // Figure out when this task can proceed again, and until then, keep
+    // yielding.
+    //
+    gettimeofday(&deadline, NULL);
+    deadline.tv_usec += (suseconds_t) lround((secs - trunc(secs)) * 1.0e6);
+    if (deadline.tv_usec > 1000000) {
+        deadline.tv_sec++;
+        deadline.tv_usec -= 1000000;
+    }
+    deadline.tv_sec += (time_t) trunc(secs);
 
-        do {
+    do {
+        if (shep == NO_SHEPHERD) {
```

Comment:
Can't we now just call chpl_task_yield() for both cases here, and let chpl_task_yield() take of whether the task is running inside a qthread?

I think that might also make the sleep implementation the same for qthreads and fifo?